Diabetes/Startup/APHAppDelegate.m

```diff
@@ -520,9 +599,382 @@ - (NSInteger)fitnessDaysShowing:(APHFitnessDaysShows)showKind
+                                                                                                            launchDateAnchor:LaunchDate];
+    
+    NSArray*            motionColumnNames   = @[@"startTime",@"activityType",@"confidence"];
+    APCPassiveDataSink* receiver            = [[APCPassiveDataSink alloc] initWithIdentifier:@"motionActivityCollector"
+                                                                                 columnNames:motionColumnNames
+                                                                          operationQueueName:@"APCCoreMotion Activity Collector"
+                                                                            andDataProcessor:CoreMotionDataSerializer];
+    
+    [motionCollector setReceiver:receiver];
+    [motionCollector setDelegate:receiver];
+    [motionCollector start];
+    [self.passiveDataCollector addDataSink:motionCollector];
+}
+
+- (void)configureObserverQueries
+{
+    NSDate* (^LaunchDate)() = ^
+    {
+        APCUser*    user        = ((APCAppDelegate *)[UIApplication sharedApplication].delegate).dataSubstrate.currentUser;
+        NSDate*     consentDate = nil;
+        
+        if (user.consentSignatureDate)
+        {
+            consentDate = user.consentSignatureDate;
+        }
+        else
+        {
+            NSFileManager*  fileManager = [NSFileManager defaultManager];
+            NSString*       filePath    = [[self applicationDocumentsDirectory] stringByAppendingPathComponent:@"db.sqlite"];
+            
+            if ([fileManager fileExistsAtPath:filePath])
+            {
+                NSError*        error       = nil;
+                NSDictionary*   attributes  = [fileManager attributesOfItemAtPath:filePath error:&error];
+                
+                if (!attributes)
+                {
+                    if (error)
+                    {
+                        APCLogError2(error);
+                        
+                        consentDate = [[NSDate date] startOfDay];
+                    }
+                }
+                else
+                {
+                    consentDate = [attributes fileCreationDate];
+                }
+            }
+        }
+        
+        return consentDate;
+    };
+    
+    NSString*(^QuantityDataSerializer)(id) = ^NSString*(id dataSample)
+    {
+        HKQuantitySample*   qtySample           = (HKQuantitySample *)dataSample;
+        NSString*           startDateTimeStamp  = [qtySample.startDate toStringInISO8601Format];
+        NSString*           endDateTimeStamp    = [qtySample.endDate toStringInISO8601Format];
+        NSString*           healthKitType       = qtySample.quantityType.identifier;
+        NSString*           quantityValueRep    = [NSString stringWithFormat:@"%@", qtySample.quantity];
+        NSArray*            valueSplit          = [quantityValueRep componentsSeparatedByString:@" "];
```

Comment:
Why are we taking the quantity sample's *description* and parsing it? Is this an attempt to recover the user's current unit preference? Instead, you should specify what units you want to use for each quantity type you are recovering, obtain the double value for that unit, and report that unit.